tumblr-photo-video-ripper.py

```diff
@@ -206,20 +222,38 @@ def _download_media(self, site, medium_type, start):
                         # if post has photoset, walk into photoset for each photo
                         photoset = post["photoset"]["photo"]
                         for photo in photoset:
-                            self.queue.put((medium_type, photo, target_folder))
+                            self.queue.put((site, medium_type, photo, target_folder))
                     except:
                         # select the largest resolution
                         # usually in the first element
-                        self.queue.put((medium_type, post, target_folder))
+                        self.queue.put((site, medium_type, post, target_folder))
                 start += MEDIA_NUM
-            except KeyError:
-                break
+                retry_times = 0
+                retry_times_for_site = 0
+            except (KeyboardInterrupt, SystemExit):
+                raise
             except UnicodeDecodeError:
-                print("Cannot decode response data from URL %s" % media_url)
-                continue
+                try:
+                    print("For site %s, Cannot decode response data from URL %s" % (site, media_url))
+                except:
+                    print("For site %s, Cannot decode response data from URL" % site)
+                break_or_continue, retry_times_for_site, retry_times, start = retry_for_parse_errors(
+                    retry_times_for_site, retry_times, start)
+                if break_or_continue:
+                    break
+                else:
+                    continue
             except:
```

Comment:
This except is the same with except `UnicodeDecodeError`.

Reply:
The unicodedecodeerror is a known error that can occur so the message that is printed is more explicit about why it failed

Reply:
We can raise an exception in `unicodedecodeerror`. And print that error message in the last except.